2022 q2 innov/carlos angelo rayel/hr 1860 summary of schedule

clrayel opened the pull request.

clrayel committed:
changed sdk version with pub get

clrayel committed:
Folder and bloc creation and pub adjutsments

clrayel committed:
Merge branch 'develop' into 2022_Q2INNOV/CARLOS_ANGELO_RAYEL/HR_1860_SUMMARY_OF_SCHEDULE

clrayel committed:
Updated pubspec.yaml

clrayel committed:
Ran build runner conflicting output cleaning

clrayel committed:
Created schedule item entity and schedule items entity

clrayel committed:
added load state (untested)

clrayel committed:
Created a test file for schedule

clrayel committed:
Fixed schedule entity test

clrayel committed:
Implemented the schedule page (ROUGH)

REF - mocked entities
REF - schedule routes
REF - schedule_page into schedule_home_page
ADD - proper (WIP) schedules page
MOD - weekly schedule link into home page drawer (under name)

clrayel committed:
Infinite loading screen with bloc

removed increment

clrayel committed:
Create schedule_mocked_entities.dart

clrayel committed:
Fixed bloc so that it actually changes from initial state

clrayel committed:
Implemented dynamic month assignment

clrayel committed:
UI refinement according to figma model

clrayel committed:
Code cleaning

clrayel committed:
Created home screen mockup

ADD home card entity
ADD home card widget
ADD home header widget
REF renamed schedule item to schedule item widget

clrayel committed:
Implemented bottom portion of login screen

clrayel committed:
Fancy summary thing on schedule page

clrayel committed:
Implemented pagination

clrayel committed:
Added shimmer as loading transition

dotlenter commented on lib/core/mocked_entities/home/home_mocked_entities.dart:
```diff
@@ -1,6 +1,6 @@
 import 'package:flutter/material.dart';
 import 'package:ui_collection/modules/home/domain/entity/home_item_entity.dart';
```
Fix imports, make it relative

dotlenter commented on lib/core/mocked_entities/schedule_mocked_entities/schedule_mocked_entities.dart:
```diff
@@ -0,0 +1,179 @@
+import 'package:flutter/material.dart';
+import 'package:ui_collection/modules/schedule/domain/entities/schedule_item_entity.dart';
```
Fix imports, make it relative

dotlenter commented on lib/core/mocked_entities/schedule_mocked_entities/schedule_mocked_entities.dart:
```diff
@@ -0,0 +1,179 @@
+
+    List<ScheduleItemEntity> week = [];
+    DateTime temp = DateUtils.dateOnly(firstDay);
+    for (int i = 0; i < 7; i++) {
```
Design rule #3, avoid loops. Use iterators

dotlenter commented on lib/modules/schedule/domain/bloc/schedule_bloc.dart:
```diff
@@ -3,34 +3,45 @@ import 'dart:async';
 import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:injectable/injectable.dart';
+import 'package:intl/intl.dart';
+import 'package:ui_collection/core/mocked_entities/schedule_mocked_entities/schedule_mocked_entities.dart';
```
Fix imports, make it relative

dotlenter commented on lib/modules/schedule/domain/bloc/schedule_bloc.dart:
```diff
@@ -3,34 +3,45 @@ import 'dart:async';
+        ScheduleMockedEntities.getItems(thisDate);
 
-    value += 1;
+    String monthYear() {
```
Function / Method Rule #5, Always have one purpose per function. 

Move this function outside

dotlenter commented on lib/modules/schedule/domain/bloc/schedule_bloc.dart:
```diff
@@ -3,34 +3,45 @@ import 'dart:async';
-      count: value,
-    ));
+    emit(_Loaded(
+        monthYear: monthYear(), currDate: thisDate, schedules: schedules));
```
Add comma to make it readable

dotlenter commented on lib/modules/schedule/domain/entities/home_card_entity.dart:
```diff
@@ -0,0 +1,10 @@
+
+@freezed
+class HomeCardEntity with _$HomeCardEntity {
+  factory HomeCardEntity({required IconData menuIcon, required String title}) =
```
Add comma to make it readable

dotlenter commented on lib/modules/schedule/domain/entities/home_cards_entity.dart:
```diff
@@ -0,0 +1,13 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ui_collection/modules/schedule/domain/entities/home_card_entity.dart';
```
Fix import, make it relative

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:ui_collection/modules/schedule/domain/bloc/schedule_bloc.dart';
```
Fix import, make it relative

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+}
+
+class _SchedulePageState extends State<ScheduleHomePage> {
+  ScheduleItemEntity getToday(List<ScheduleItemEntity> schedules) =>
```
Method Rule #3, Always place similar or related functions next to each other.

Move this method after the build method.

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+            ),
+            preferredSize: Size.fromHeight(1),
+          ),
+          leading: Builder(builder: (context) {
```
Make a callback function for this.

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+                color: CommonColors.green[300],
+              ),
+            );
+          })),
```
Add comma at the end to make it readable.

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+            );
+          })),
+      drawer: AppDrawer(),
+      body: BlocProvider(create: (BuildContext context) {
```
Instead of `create: (BuildContext context { return; }` make it shorthand:

`create: (_) => ScheduleBloc,`

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+        return ScheduleBloc();
+      }, child: BlocBuilder<ScheduleBloc, ScheduleState>(
+        builder: (context, state) {
+          return state.when(initial: () {
```
remove mapping of `initial` state. Move this to initState to fully use the `StatefulWidget`

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+            return const Center(
+              child: CircularProgressIndicator(),
+            );
+          }, loaded: (String monthYear, DateTime currDate,
```
Create callback function for this

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+                )
+              ],
+            );
+          }, loading: () {
```
Create a callback function for this

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+                    ),
+                  ),
+                ),
+                Container(
```
Remove unecessary containers

dotlenter commented on lib/modules/schedule/page/schedule_home_page.dart:
```diff
@@ -0,0 +1,164 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:ui_collection/modules/schedule/domain/bloc/schedule_bloc.dart';
+import 'package:ui_collection/modules/schedule/widget/drawer/app_drawer_widget.dart';
```
Fix imports, make it relative.

dotlenter commented on lib/modules/schedule/page/schedule_page.dart:
```diff
@@ -1,5 +1,12 @@
 import 'package:flutter/material.dart';
-import 'package:ui_collection/modules/schedule/widget/drawer/app_drawer_widget.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
```
Fix imports, make it relative

dotlenter commented on lib/modules/schedule/page/schedule_page.dart:
```diff
@@ -11,31 +18,126 @@ class SchedulePage extends StatefulWidget {
-                      fontSize: 34,
-                    ),
+    return BlocProvider(
+      create: (context) => ScheduleBloc(),
```
Make this shorthand

dotlenter commented on lib/modules/schedule/page/schedule_page.dart:
```diff
@@ -11,31 +18,126 @@ class SchedulePage extends StatefulWidget {
+            )),
+        body: BlocBuilder<ScheduleBloc, ScheduleState>(
+          builder: (context, state) {
+            return state.when(initial: () {
```
Remove `initial` state and Move the `initial` state function to **initState**

dotlenter commented on lib/modules/schedule/page/schedule_page.dart:
```diff
@@ -11,31 +18,126 @@ class SchedulePage extends StatefulWidget {
+              BlocProvider.of<ScheduleBloc>(context)
+                  .add(ScheduleEvent.load(currDate: DateTime.now()));
+              return LoadingScreenWidget();
+            }, loaded: (String monthYear, DateTime currDate,
```
Create a callback function for loaded

dotlenter commented on lib/modules/schedule/page/schedule_page.dart:
```diff
@@ -11,31 +18,126 @@ class SchedulePage extends StatefulWidget {
+                      ))
+                ],
+              );
+            }, loading: () {
```
Create a callback function for loading

dotlenter commented on lib/modules/schedule/page/schedule_page.dart:
```diff
@@ -11,31 +18,126 @@ class SchedulePage extends StatefulWidget {
+                  Expanded(
+                      flex: 9,
+                      child: ListView.separated(
+                        itemBuilder: (context, index) => ScheduleItem(
```
Create a callback function for itemBuilder

dotlenter commented on lib/modules/schedule/page/schedule_page.dart:
```diff
@@ -11,31 +18,126 @@ class SchedulePage extends StatefulWidget {
+                        itemBuilder: (context, index) => ScheduleItem(
+                          schedule: schedules[index],
+                        ),
+                        separatorBuilder: (context, index) => const Divider(
```
Create a callback function

dotlenter commented on lib/modules/schedule/widget/drawer/user_details_widget.dart:
```diff
@@ -50,6 +52,28 @@ class UserDetailsWidget extends StatelessWidget {
+                          style: TextStyle(
+                            color: CommonColors.green,
+                            fontSize: 14,
+                          )),
```
Add comma to make it readable

dotlenter commented on lib/modules/schedule/widget/drawer/user_details_widget.dart:
```diff
@@ -50,6 +52,28 @@ class UserDetailsWidget extends StatelessWidget {
+                            color: CommonColors.green,
+                            fontSize: 14,
+                          )),
+                    )),
```
Add comma to make it readable

dotlenter commented on lib/modules/schedule/widget/home_widgets/clocking_widget.dart:
```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+import 'package:ui_collection/widget/colors/common_color.dart';
```
Fix import, make it relative

dotlenter commented on lib/modules/schedule/widget/home_widgets/clocking_widget.dart:
```diff
@@ -0,0 +1,62 @@
+  const ClockingWidget({Key? key, required this.title, required this.icon})
+      : super(key: key);
+
+  ButtonStyle getButtonStyle(String title) {
```
make this function private by putting a `_` prefix in the function name example: `_getButtonStyle`

dotlenter replied:
Move these functions after `build` method

dotlenter commented on lib/modules/schedule/widget/home_widgets/clocking_widget.dart:
```diff
@@ -0,0 +1,62 @@
+    return ButtonStyle();
+  }
+
+  Color getIconColor(String title) {
```
Make this function private

dotlenter commented on lib/modules/schedule/widget/home_widgets/header_widget.dart:
```diff
@@ -0,0 +1,32 @@
+import 'package:flutter/material.dart';
+import 'package:ui_collection/modules/schedule/domain/entities/schedule_item_entity.dart';
```
Fix import

dotlenter commented on lib/modules/schedule/widget/home_widgets/clocking_widget.dart:
```diff
@@ -0,0 +1,62 @@
+            style: TextStyle(color: CommonColors.neutral),
+          ),
+          SizedBox(),
+        ]),
```
Add comma to make it readable

dotlenter commented on lib/modules/schedule/widget/home_widgets/menu_card_widget.dart:
```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/cupertino.dart';
+import 'package:ui_collection/modules/schedule/domain/entities/home_card_entity.dart';
```
Fix imports

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+import 'package:flutter/material.dart';
+import 'package:ui_collection/core/mocked_entities/schedule_mocked_entities/schedule_mocked_entities.dart';
```
Fix imports

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+                      icon: Icon(
+                        Icons.keyboard_arrow_left,
+                        color: CommonColors.green,
+                      )),
```
Add comma

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+                      icon: Icon(
+                        Icons.keyboard_arrow_right,
+                        color: CommonColors.green,
+                      )),
```
Add comma

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+                  ),
+                ],
+              ),
+            )),
```
Add comma

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+              ),
+              itemCount: ScheduleMockedEntities.mockWeek.length,
+              shrinkWrap: true,
+            ))
```
Add comma

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+        Expanded(
+            flex: 9,
+            child: ListView.separated(
+              itemBuilder: (context, index) => ShimmerLoading(
```
Create a callback function for itemBuilder

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+                  schedule: ScheduleMockedEntities.mockWeek[index],
+                ),
+              ),
+              separatorBuilder: (context, index) => const Divider(
```
create a callback function

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/loading_screen_widget.dart:
```diff
@@ -0,0 +1,126 @@
+    );
+  }
+
+  Widget AbbrBlock() => Container(
```
Always write in camel case format. (lowerCamelCase)
Always write function or method name with these formats
    - verb followed by a noun.
    - preposition followed by a noun then a verb.

dotlenter commented on lib/modules/schedule/widget/loading_screen_widget/schedule_item_shimmer_widget.dart:
```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+import 'package:ui_collection/widget/colors/common_color.dart';
```
Fix import

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_item_widget.dart:
```diff
@@ -0,0 +1,106 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:ui_collection/widget/colors/common_color.dart';
```
Fix import

clrayel committed:
Layout bug fixing

Set height of all container widgets for schedule page except the listview at the bottom, which is expanded

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_item_widget.dart:
```diff
@@ -0,0 +1,106 @@
+
+  const ScheduleItem({Key? key, required this.schedule}) : super(key: key);
+
+  getItemColor(DateTime dateTime, String status) {
```
Make this function private, move this after build method

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_item_widget.dart:
```diff
@@ -0,0 +1,106 @@
+    return CommonColors.neutral[500];
+  }
+
+  getStatusDisplay(String status) {
```
Make this method private

dotlenter replied:
Move this after `build` method

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_summary_widget.dart:
```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:ui_collection/widget/colors/common_color.dart';
```
Fix import

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_summary_widget.dart:
```diff
@@ -0,0 +1,103 @@
+  const ScheduleSummaryWidget({Key? key, required this.schedule})
+      : super(key: key);
+
+  Color getTextColor(DateTime dateTime, String status) {
```
Make this private

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_summary_widget.dart:
```diff
@@ -0,0 +1,103 @@
+    return CommonColors.neutral[500]!;
+  }
+
+  Color getTileColor(DateTime dateTime, String status) {
```
Make this private

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_summary_widget.dart:
```diff
@@ -0,0 +1,103 @@
+    if (dateTime.year == DateTime.now().year &&
+        dateTime.month == DateTime.now().month &&
+        dateTime.day == DateTime.now().day) {
+      if (status != "Normal Shift" && status != "Rest Day") {
```
Avoid nested if statements

dotlenter commented on lib/modules/schedule/widget/schedule_widgets/schedule_summary_widget.dart:
```diff
@@ -0,0 +1,103 @@
+    return CommonColors.neutral[0]!;
+  }
+
+  Widget getDayWidget(DateTime dateTime, String status) {
```
Make this private, Move this after build method

dotlenter commented on test/modules/schedule/domain/bloc/schedule_test.dart:
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
```
When testing a class file, copy the file name and add the suffix "_test".

Copy the directory as well

clrayel committed:
Review changes
